src/encore.ts

```diff
@@ -0,0 +1,152 @@
+import * as vscode from "vscode";
+import {
+    WorkspaceFolder,
+    DebugConfiguration,
+    ProviderResult,
+    CancellationToken,
+} from "vscode";
+import * as os from "os";
+import * as path from "path";
+import { MessageType, getEncoreId, showMessage } from "./utils";
+import { getPlatformDebugEncoreProcess } from "./utils/process-utils";
+
+export class EncoreDebugConfigurationProvider
+    implements vscode.DebugConfigurationProvider
+{
+    resolveDebugConfiguration(
+        folder: WorkspaceFolder | undefined,
+        config: DebugConfiguration,
+    ): ProviderResult<DebugConfiguration> {
+        // delegate to existing provider.
+        launchEncoreDebugSession(folder, { noDebug: config.noDebug });
+        return undefined; // Cancel this debug session
+    }
+}
+
+export function manualLaunchDebugSession() {
+    if (!vscode.workspace.workspaceFolders) {
+        showMessage("No active workspace folders");
+        return;
+    }
+    launchEncoreDebugSession(vscode.workspace.workspaceFolders[0], {
+        noDebug: false,
+    });
+}
+
+export function getEncoreGoOverrideFilePath(): string {
```

Comment:
Would be good if this script was bundled with the extension, as not everyone will have it, or have it in the same location.

Reply:
Hmm, yeah, I get your concern. I can do that. On second thought, what about having it inside .vscode? I'm thinking that even if we bundle it with the extension, if, for some reason, a user doesn't use the extension and .vscode is committed to git by another user who uses the extension, vscode.go will use the setting and assume the file exists, as there is no way to ensure that it exist like we currently do with the extension.